### PR TITLE
geolocation viewport for pieplot api

### PR DESCRIPTION
**Pull request by d-callan**

### CODE REVIEW

File: schema/url/pass/plugin-pieplot.raml [hunk 1]
```diff
@@ -18,12 +18,19 @@ types:
...
       valueSpec: 
         type: ValueSpec
+      longitudeVariable?:
+        type: VariableSpec
+      latitudeVariable?:
+        type: VariableSpec
+      viewport?:
+        type: GeolocationViewport
 
   PieplotConfig:
```
Comment (asizemore) [inline schema/url/pass/plugin-pieplot.raml:35]: looks like plot.data is returning a viewport as part of the pieplot$config. Should be added here as well?

**Review by asizemore — COMMENTED** [review]
🥧

File: src/main/java/org/veupathdb/service/eda/ds/plugin/AbstractPlugin.java [hunk 2]
```diff
@@ -321,6 +322,17 @@ protected String getViewportAsRString(NumericViewport viewport, String xVarType)
     }
   }
 
+  protected String getViewportAsRString(GeolocationViewport viewport) {
```
[the remark this answers is not in the document]
Comment (d-callan) [reply src/main/java/org/veupathdb/service/eda/ds/plugin/AbstractPlugin.java:325]: its an overloaded method. there should be a second that takes NumericViewport or whatever weve called it.

**Review by asizemore — APPROVED** [review]
lovely!

**Pull request merged**